src/github.com/getlantern/lantern-android/client/config.go

```diff
@@ -0,0 +1,122 @@
+package client
+
+import (
+	"compress/gzip"
+	"crypto/x509"
+	"errors"
+	"github.com/getlantern/fronted"
+	"github.com/getlantern/keyman"
+	"github.com/getlantern/yaml"
+	"io"
+	"io/ioutil"
+	"net/http"
+)
+
+type clientCfg struct {
+	FrontedServers []frontedServer                  `yaml:"frontedservers"`
+	MasqueradeSets map[string][]*fronted.Masquerade `yaml:"masqueradesets"`
+}
+
+// Config provides client configuration.
+type Config struct {
+	Client     clientCfg `yaml:"client"`
+	TrustedCAs []*CA     `yaml:"trustedcas"`
+}
+
+var (
+	// ErrFailedConfigRequest is returned when the server replies with a non-200
+	// status code to our request for a configuration file.
+	ErrFailedConfigRequest = errors.New(`Could not get configuration file.`)
+
+	// ErrInvalidConfiguration is returned in case the configuration file is
+	// downloaded but has no useful data.
+	ErrInvalidConfiguration = errors.New(`Invalid configuration file.`)
+)
+
+const (
+	// URL of the configuration file. Remember to use HTTPs.
+	remoteConfigURL = `https://s3.amazonaws.com/lantern_config/cloud.1.6.0.yaml.gz`
+)
+
+// pullConfigFile attempts to retrieve a configuration file over the network,
+// then it decompresses it and returns the file's raw bytes.
+func pullConfigFile() ([]byte, error) {
+	var err error
+	var res *http.Response
+
+	// Issuing a post request to download configuration file.
+	if res, err = http.Get(remoteConfigURL); err != nil {
```

Comment:
This will rarely work in the censored world.  We have to assume that the site from which we're fetching the updated configuration is blocked.  In fact, the site we use right now is s3.amazonaws.com, and that is periodically blocked in China.  That's why flashlight uses proxy for the http client - https://github.com/getlantern/flashlight-build/blob/peerscanner/src/github.com/getlantern/flashlight/config/config.go#L290

What flashlight actually does is the following:

1. Start using a list of servers and masquerades that's compiled into the binary (I see that you have that list, but aren't actually starting with it unless you're unable to fetch the list from the cloud)
2. Periodically fetch updates from the cloud by making proxied http requests (I see that you're only checking once, which is okay for now)
3. Update its configuration whenever a change has been detected in the cloud
4. Save the configuration to disk for whenever it next starts up (it looks like you're not saving it anywhere?)

The idea behind this is that the binary that we distribute will work for a while, but we may have to change our set of masquerades and servers to stay ahead of the censors.  As long as we're actually staying ahead of the censors, this will keep any Lantern clients that are already running unblocked.  And since these clients save their configuration to disk, they should remain unblocked even after a restart.

Ultimately, I think we'll want to do something similar for the Android version, but we don't need to be that sophisticated yet.  I think the way that you have should be fine with two changes:

1. Immediately start proxying based on the built-in configuration
2. Check for the cloud configuration through the already running proxy